Test: import old docs

GabiNecasova opened the pull request:
- This tries to import Markdown files from [docs.podman.io](docs.podman.io) to the [podman.io](podman.io)
- This is just a test, it means almost no modifications of Markdown files, the goal is just to check how does it looks like on podman.io page
- Note: if a command e.g. `podman generate` has subcommands ( `podman generate systemd`, `podman generate`), then we can create a sidebar category for easier navigation (similar to [Docker](https://docs.docker.com/engine/reference/commandline/cli/)) 
- Note: The "SEE ALSO" sections of Markdown files are now commented because they cause link error

![Screenshot from 2023-11-15 17-58-23](https://github.com/containers/podman.io/assets/141641852/42263cb9-dcbe-4d2e-96bc-eddec75447bc)
![Screenshot from 2023-11-15 17-58-00](https://github.com/containers/podman.io/assets/141641852/f273392a-2446-48b1-8dcf-09df8cb425ca)

TomSweeneyRedHat commented on getting-started/installation.md:
```diff
@@ -213,6 +213,21 @@ There are many [packages](https://packages.ubuntu.com/search?keywords=libpod&sea
 
+#### [Ansible](https://www.ansible.com/)
+
+An [Ansible Role](https://github.com/alvistack/ansible-role-podman) is also available to automate the above steps:
```
This doesn't work for MacOS and kind of implies that it does.  Maybe
```suggestion
An [Ansible Role](https://github.com/alvistack/ansible-role-podman) is also available to automate the above steps for many of the distributions.  For a full list of the distributions, please reference the [Ansible Role Requirements](https://github.com/alvistack/ansible-role-podman#requirements).
```

hswong3i replied:
I change into this:
> An Ansible Role is also available to automate the installation of above statically linked binary on its supported OS:

TomSweeneyRedHat commented on getting-started/installation.md:
```diff
@@ -545,6 +545,43 @@ Note that Podman does not officially support device-mapper. Thus, the `exclude_g
 
+### Static build
+
+It is possible to build a statically linked binary of podman by using the
```
```suggestion
It is possible to build a statically linked binary of Podman by using the
```

hswong3i replied:
Fixed

TomSweeneyRedHat commented on getting-started/installation.md:
```diff
@@ -545,6 +545,43 @@ Note that Podman does not officially support device-mapper. Thus, the `exclude_g
+#### Ansible
+
+An [Ansible Role](https://github.com/alvistack/ansible-role-podman) is
+also available to automate the installation of above statically linked
```
```suggestion
also available to automate the installation of the above statically linked
```

hswong3i replied:
Fixed

hswong3i replied:
@TomSweeneyRedHat thank you very much -;)

hswong3i commented on getting-started/installation.md:
```diff
@@ -545,6 +545,43 @@ Note that Podman does not officially support device-mapper. Thus, the `exclude_g
+
+#### Nix
+
+To build the binaries by locally install the nix package manager:
```
Fixed

TomSweeneyRedHat commented:
Thanks @hswong3i .  Is everything in place on the Podman side to merge this once we get the requisite LGTM's?

@mheon, @rhatdan WDYT?

TomSweeneyRedHat commented:
@hswong3i in particular, does https://github.com/containers/libpod/pull/6402/ need to merge before this merges?

hswong3i commented:
Basically nix support already in libpod’s master; btw I guess we better wait for https://github.com/containers/libpod/pull/6402 ;-)

rhatdan commented:
@hswong3i What is the latest on this one?

hswong3i commented:
@rhatdan both https://github.com/containers/libpod/pull/6408 and https://github.com/containers/libpod/pull/6402 are ready for review, where statically linked binary with nix-based are compile correctly.

I also upload those statically linked binary to https://github.com/alvistack/libpod/releases/download/v1.9.3/podman-v1.9.3-linux-amd64 where linking my Ansible Role to there; therefore from this point of view the Ansible Role itself is ready for daily use, or include into podman.io as alternative installation method ;-)

TomSweeneyRedHat commented:
@hswong3i I think this is good to merge now based on the progress with the noted PRs.  Do you concur?

hswong3i commented:
@TomSweeneyRedHat currently https://github.com/containers/podman/pull/6402 almost passing required CI

On the other hand, I keep uploading my local build for static binary to my fork (e.g. https://github.com/alvistack/podman/releases/tag/v2.0.2) and download it by Ansible Role, so this PR should be good enough for merge?

mbbroberg commented on getting-started/installation.md:
````diff
@@ -545,6 +545,43 @@ Note that Podman does not officially support device-mapper. Thus, the `exclude_g
+
+``` shell
+curl -L https://nixos.org/nix/install | sh
+git clone https://github.com/containers/libpod.git && cd libpod
````
FWIW, I had trouble adding nix to my path when running through this (running F32). The script recommends running `. /home/mbbroberg/.nix-profile/etc/profile.d/nix.sh`, and I see nix is found in `~/.nix-profile/bin/nix`. It might be helpful to add that as a separate step. To make it work in any shell, it's worth pointing out the latter to the reader.

```suggestion
# The nix binary will be at ~/.nix-profile/bin/nix
# Add it to the path manually or run the following:
. /home/mbbroberg/.nix-profile/etc/profile.d/nix.sh
git clone https://github.com/containers/libpod.git && cd libpod
```

hswong3i replied:
Different nix setup may coming with different result... Here I just going to provide a hints for general setup :-S

mbbroberg commented on getting-started/installation.md:
```diff
@@ -545,6 +545,43 @@ Note that Podman does not officially support device-mapper. Thus, the `exclude_g
+curl -L https://nixos.org/nix/install | sh
+git clone https://github.com/containers/libpod.git && cd libpod
+nix build -f nix/
+./result/bin/podman --version
```
Looks like this may have changed.

```
./result-bin/bin/podman --version
podman version 2.1.0-dev
```

```suggestion
./result-bin/bin/podman --version
```

hswong3i replied:
This PR changes is based on https://github.com/containers/podman/pull/6402, once that merged path will become `./result/bin/podman`

mbbroberg commented on getting-started/installation.md:
```diff
@@ -545,6 +545,43 @@ Note that Podman does not officially support device-mapper. Thus, the `exclude_g
+cd ~/.ansible/roles/podman
+pip3 install --upgrade --ignore-installed --requirement requirements.txt
+molecule converge
+molecule verify
```
This all worked up to the molecule part. I'm erroring on a sudo requirement for Python (?). Probably user error but sharing in case it's helpful context. 

```
fatal: [localhost]: FAILED! => {
    "ansible_facts": {},
    "changed": false,
    "failed_modules": {
        "setup": {
            "ansible_facts": {
                "discovered_interpreter_python": "/usr/bin/python3"
            },
            "failed": true,
            "module_stderr": "sudo: a password is required\n",
            "module_stdout": "",
            "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
            "rc": 1
        }
    },
    "msg": "The following modules failed to execute: setup\n"
}
    
    PLAY RECAP *********************************************************************
    localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
    
ERROR: 
```

mbbroberg commented:
Hey all, newer podman user and wanted to offer feedback and help in this repo where I can. Lmk if you need any follow up.

rhatdan commented:
@mbbroberg Welcome, most of the work is being done in https://github.com/containers/podman but docs work can be handled here.  Any help is appreciated.

hswong3i commented:
With https://github.com/containers/podman.io/commit/062c1b13daf353481e01b6d1c0e8d1c092ade31d I also try to regex replace all legacy URL from https://github.com/containers/libpod into https://github.com/containers/podman

mbbroberg commented:
I tested some of the links and confirm the regex worked alright @hswong3i.

rhatdan commented:
LGTM

TomSweeneyRedHat commented:
There's one small issue with a readmore tag being removed.  However, the rest LGTM.  So I think I'm going to merge this now and spin up a new PR to re-add the readmore tag.  @hswong3i @mbbroberg TYVM for your work on this!

GabiNecasova committed:
Test: import old docs

- this tries to import Markdown files from docs.podman.io to the
  podman.io
- this is just a test, it means almost no modifications of Markdown
  files, the goal is just to check how does it looks like on podman.io

Signed-off-by: Gabriela Necasova <gnecasov@redhat.com>

Luap99 commented:
> This tries to import Markdown files from [docs.podman.io](https://github.com/containers/podman.io/pull/docs.podman.io) to the [podman.io](https://github.com/containers/podman.io/pull/podman.io)

Not that I am against this but there is one big thing to consider and I am not sure if you are aware of it so will write it down here again:
The docs are based of https://github.com/containers/podman/tree/main/docs/source and it is hard requirement that the source must live in the podman repo, when we make code changes over there we have the requirement that the docs must be updated at the same time. Trying to do this over more than one repo is simply unsustainable and would complicate the work considerably.
Right now the docs are build and published by readthedocs (https://readthedocs.org/projects/podman/). They will automatically update for each merge and provide build for each version. This is important functionality that we would need to replicate if we want this under podman.io/docs. And to me this looks like a lot of work to reinvent readthedocs functionality.


So my first question is what benefit does podman.io/docs has over docs.podman.io? Is it really worth all the effort to solve the requirements mentioned above. If this is only about style then would it not be simpler to change the theme for sphinx build https://github.com/containers/podman/blob/main/docs/source/conf.py#L63-L75 and/or add custom css there?

GabiNecasova commented:
@Luap99 Thank you very much for your comments. I didn't know exactly about this complication and I understand. In that case, it is indeed probably the best solution to keep the documentation in the Podman repository and just change the visual style (I think the dark mode is very useful).

rhatdan commented:
@GabiNecasova still working on this?

GabiNecasova commented:
@rhatdan No, currently I am reviewing the podman.io content and I will make a couple of suggestions. 
I think I can close this issue. Thank you for pointing this out.

GabiNecasova commented:
I am closing this.
